CurrencyExchangeServiceWCF/CurrencyExchangeServiceWCF/Service.svc.cs

```diff
@@ -1,81 +1,100 @@
+    public class Service1 : IService1
     {
-        public string ConvertCurrency(string toCurrency, string value)
+        SoapException soapException ;
+        public Response<string> ConvertCurrency(ConvertCurrenyRequest req)
         {
-            string currencyInputPattern = @"^[a-zA-Z]{3}$";
-            string valueInputPattern = @"^[,.0-9]*$";
+            try
+            {
+                Response<string> r = new Response<string>();
+                if (!req.autHeader.password.Equals("pa$$w0rd") || !req.autHeader.username.Equals("Admin")){
+                    r.ReturnValue = "authentication not valid"; 
+                    return r;
+                }
+                String toCurrency = req.toCurrency;
+                String value = req.ccValue;
+                toCurrency = toCurrency.ToUpper();
 
-           
+                string currencyInputPattern = @"^[a-zA-Z]{3}$";
+                string valueInputPattern = @"^[,.0-9]*$";
 
-            if (String.IsNullOrEmpty(value) || String.IsNullOrEmpty(toCurrency))
-            {
-                return "Input cannot be NULL!";
-            }
-            toCurrency = toCurrency.ToUpper();
 
 
-            bool isCurrencyValid = Regex.IsMatch(toCurrency, currencyInputPattern);
-            bool isValueValid = Regex.IsMatch(value, valueInputPattern);
+                if (String.IsNullOrEmpty(value) || String.IsNullOrEmpty(toCurrency))
+                {
+                    r.ReturnValue = "Input cannot be NULL!";
+                    return r;
+                }
+                toCurrency = toCurrency.ToUpper();
 
-            if (!isCurrencyValid || !isValueValid)
-            {
-                return "Invalid input!";
-            }
 
-            // necessary because the European Central Bank only shows the exchange rate list based in EUR, that is why we have to workaround and have our base in USD
-            if (toCurrency.Equals("USD"))
-            {
-                return value;
-            }
-            if (toCurrency.Equals("EUR"))
-                toCurrency = "USD";
+                bool isCurrencyValid = Regex.IsMatch(toCurrency, currencyInputPattern);
+                bool isValueValid = Regex.IsMatch(value, valueInputPattern);
 
-            if (!GetCurrencyCodes().Contains(toCurrency))
-            {
-                return "Currency not available in the CurrencyConverter currencies list.";
+                if (!isCurrencyValid || !isValueValid)
+                {
+                    r.ReturnValue = "Input cannot be NULL!";
+                    return r;
+                }
 
-            }
+                double ConversionRateDouble;
+                string ConversionRateString;
+            
+                if (toCurrency.Equals("USD"))
+                {
+                    r.ReturnValue = value;
+                    return r;
+                }
+                if (toCurrency.Equals("EUR"))
+                    toCurrency = "USD";
```

Comment:
Is this correct? This is a hardcoded change of the incoming parameter

Reply:
this is a algotithmus from Christian, i gues he will map the EUR to USD or EUR = USD

Reply:
I guess this is leagcy code then. but leave it for now